src/core/ext/filters/client_channel/resolving_lb_policy.cc

```diff
@@ -248,27 +268,46 @@ void ResolvingLoadBalancingPolicy::OnResolverShutdownLocked(grpc_error* error) {
               resolver_.get());
     }
     resolver_.reset();
-    grpc_error* error = GRPC_ERROR_CREATE_REFERENCING_FROM_STATIC_STRING(
-        "Resolver spontaneous shutdown", &error, 1);
+    grpc_error* error =
```

Comment:
Regarding the comment above about this code being defensive, because:
a) This request router holds a ref on the resolver
b) `OnResolverShutdownLocked` is called from the request router's dtor
c) The request router is a <i>private</i> `UniquePtr` owned by the `Resolver` base class

... isn't it now impossible to reach `OnResolverShutdownLocked` unless this request router has already orphaned the resolver?

Reply:
A buggy resolver implementation could still unref itself too many times and cause spontaneous shutdown.

Reply:
You're right.  And, in fact, I think that means that this entire method is not needed, because we always orphan the LB policy at the same time that we orphan the resolver, so there's really nothing that we need to do here.

Fixed.